### PR TITLE
feat: Longhorn toolbox CLI

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#7927

#### What this PR does / why we need it:

Add common functions to be used by Longhorn Toolbox CLI.

#### Special notes for your reviewer:

None

#### Additional documentation or context

None

### CODE REVIEW

File: longhorn/longhorn.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package longhorn
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// GetVolumeNameFromReplicaName returns the volume name from the replica name.
```
Comment (james-munson) [inline longhorn/longhorn.go:10]: Maybe it's worth mentioning that the replica name is the data directory name, not the replica CR object name (with -r-).  Might even be worth a test case for that.

**Review by james-munson — COMMENTED** [review]
Still working, but one small comment so far.

File: io/file.go [hunk 2]
```diff
@@ -164,12 +165,29 @@ func GetEmptyFiles(directory string) (filePaths []string, err error) {
 
 // FindFiles searches for files in the specified directory with the given fileName.
 // If fileName is empty, it retrieves all files in the directory.
+// If maxDepth is greater than 0, it limits the search to the specified depth.
 // It returns a slice of filePaths and an error if any.
-func FindFiles(directory, fileName string) (filePaths []string, err error) {
+func FindFiles(directory, fileName string, maxDepth int) (filePaths []string, err error) {
```
Comment (derekbit) [inline io/file.go:170]: Can we use [the 3rd-party implementation](https://github.com/Azure/draft/blob/5008e06cb5ebe6a3c54b30c40a09e34df3329ccd/pkg/reporeader/reporeader.go#L54)?

Comment (c3y1huang) [reply io/file.go:170]: > Can we use [the 3rd-party implementation](https://github.com/Azure/draft/blob/5008e06cb5ebe6a3c54b30c40a09e34df3329ccd/pkg/reporeader/reporeader.go#L54)?

This is used for unit-testing Azure/draft, I prefer not to use it. What is the benefit?

Comment (derekbit) [reply io/file.go:170]: > What is the benefit?

This is a very general function, and we don't need to implement it again and maintain it.
I'm fine with the current codes.

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@c3y1huang We can merge it after resolving the conflicts. Thank you.

**Review by derekbit — APPROVED** [review]

**Pull request merged**